lib/SILOptimizer/Mandatory/MoveKillsCopyableAddressesChecker.cpp

```diff
@@ -907,6 +1072,27 @@ cleanupAllDestroyAddr(SILFunction *fn, SmallBitVector &destroyIndices,
     madeChange = true;
   }
 
+  // Check for consuming closures. The closure can have either destroy uses or
+  // reinit uses. We need to handle both.
+  for (int index = consumingClosureIndices.find_first(); index != -1;
+       index = consumingClosureIndices.find_next(index)) {
+    auto &pair = *std::next(useState.consumingClosureOperands.begin(), index);
+    auto *closureUse = pair.first;
+    if (!closureUse)
+      continue;
+    for (auto *pairedInst : pair.second.pairedConsumingInsts) {
+      if (auto *dvi = dyn_cast<DestroyValueInst>(pairedInst)) {
+        dvi->eraseFromParent();
+        madeChange = true;
+        continue;
+      }
+
+      convertMemoryReinitToInitForm(pairedInst);
```

Comment:
NOTE: In the next commit in this PR, I change this to use the cloning logic. This was just a stop gap (that actually emits incorrect code due to the lack of conversion to the out parameter). So if one is reading commit by commit, check the next commit.